src/components/HealthCardDisplay/HealthCardDisplay.js

```diff
@@ -114,133 +114,185 @@ const HealthCardDisplay = () => {
-    topBannerStyle = styles.bannerError;
-    topBannerIcon = exclamationRedIcon;
-    topBannerText = t('healthcarddisplay.Not verified');
-  } else if (healthCardVerified.verified && issuerVerified) {
-    topBannerStyle = styles.topBannerValid;
-    topBannerIcon = checkIcon;
-    topBannerText = t('healthcarddisplay.Verified');
-    bottomBannerStyle = styles.bottomBannerValid;
-  } else {
-    topBannerStyle = styles.topBannerPartial;
-    topBannerIcon = exclamationOrangeIcon;
-    topBannerText = t('healthcarddisplay.Partially Verified');
-    bottomBannerStyle = styles.bottomBannerPartial;
-  }
+  const Banners = ({ cardJws, cardsValidPrimarySeries }) => {
+    const {
+      setJws,
+      healthCardSupported,
+      healthCardVerified,
+      issuerVerified,
+    } = useHealthCardDataContext();
+    const [topBannerStyle, setTopBannerStyle] = useState(null);
+    const [topBannerIcon, setTopBannerIcon] = useState(null);
+    const [topBannerText, setTopBannerText] = useState(null);
+    const [bottomBannerStyle, setBottomBannerStyle] = useState(null);
 
-  return (
-    <Grid container className={styles.root}>
+    useEffect(() => {
+      setJws(cardJws);
+    }, [cardJws, setJws]);
+
+    useEffect(() => {
+      if (!healthCardSupported.status) {
+        setTopBannerStyle(styles.bannerError);
+        setTopBannerIcon(exclamationRedIcon);
+        setTopBannerText(t('healthcarddisplay.Invalid SMART Health Card'));
+      } else if (!healthCardVerified.verified && healthCardVerified.error) {
+        setTopBannerStyle(styles.bannerError);
+        setTopBannerIcon(exclamationRedIcon);
+        setTopBannerText(t('healthcarddisplay.Not verified'));
+      } else if (healthCardVerified.verified && issuerVerified) {
+        setTopBannerStyle(styles.topBannerValid);
+        setTopBannerIcon(checkIcon);
+        setTopBannerText(t('healthcarddisplay.Verified'));
+        setBottomBannerStyle(styles.bottomBannerValid);
+      } else {
+        setTopBannerStyle(styles.topBannerPartial);
+        setTopBannerIcon(exclamationOrangeIcon);
+        setTopBannerText(t('healthcarddisplay.Partially Verified'));
+        setBottomBannerStyle(styles.bottomBannerPartial);
+      }
+    }, [healthCardSupported.status, healthCardVerified.error,
+      healthCardVerified.verified, issuerVerified]);
+
+    const TopBanner = ({
+      img, alt, style, text
+    }) => (
+      <Grid item xs={12} className={style} width="100%">
+        <Container style={{ width: 'fit-content' }}>
+          <Box
+            className={styles.flexRow}
+            pt={2}
+            pb={2}
+          >
+            <img
+              src={img}
+              alt={alt}
+              style={{ height: '2rem', marginRight: '1rem' }}
+            />
+            <Typography variant="h4">
+              {text}
+            </Typography>
+          </Box>
+        </Container>
+      </Grid>
+    );
+
+    const BottomBanner = ({
```

Comment:
When I scan the SHCs linked here: https://github.com/dvci/shc-web-verifier/pull/8#pullrequestreview-813727922, I'm brought to this weird error page that also displays the bottom banners:

![Screen Shot 2022-04-08 at 10 49 55 AM](https://user-images.githubusercontent.com/51176978/162490403-8b2ba6d1-f583-4e04-b4dd-f57ca9eb3d48.png)



whereas, when I scan this card on the main branch, I'm taken to the  normal "Not verified" page:

![image](https://user-images.githubusercontent.com/51176978/162490482-a477e42b-58bd-4460-8ba8-ac03895adf47.png)

Reply:
Made a bunch of changes, should be fixed now